pkg/monitor/alerts.go

```diff
@@ -381,11 +381,10 @@ func CreateEventIntervalsForAlerts(ctx context.Context, alerts prometheustypes.V
 			default:
 				level = monitorapi.Error
 			}
-			alertIntervalTemplate := monitorapi.Interval{
-				Condition: monitorapi.NewCondition(level).
-					Locator(lb).
-					Message(monitorapi.NewMessage().HumanMessage(alert.Metric.String())).Build(),
-			}
+			alertIntervalTemplate := monitorapi.NewInterval(monitorapi.IntervalCategoryAlert, level).
+				Locator(lb).
+				Message(monitorapi.NewMessage().
+					HumanMessage(alert.Metric.String())).Build()
```

Comment:
Sample of interval creation that compiles with this change, we force specification of a type.